addons/mail_plugin/controllers/mail_plugin.py

```diff
@@ -0,0 +1,290 @@
+        versions of the mail plugin but necessary for supporting older versions
+        """
+        response = {}
+
+        if not (partner_id or (name and email)):
+            return {'error': _('You need to specify at least the partner_id or the name and the email')}
+
+        if partner_id:
+            partner = request.env['res.partner'].browse(partner_id)
+            return self._prepare_contact_values(partner)
+
+        normalized_email = tools.email_normalize(email)
+        if not normalized_email:
+            response['error'] = 'Bad email.'
+            return response
+
+        # Search for the partner based on the email.
+        # If multiple are found, take the first one.
+        partner = request.env['res.partner'].search(['|', ('email', 'in', [normalized_email, email]),
+                                                     ('email_normalized', '=', normalized_email)], limit=1)
+
+        response = self._prepare_contact_values(partner)
+
+        # if no partner is found in the database, we should also return an empty one having id = -1, otherwise older versions of
+        # plugin won't work
+        if not response['partner']:
+            sender_domain = normalized_email.split('@')[1]
+            response['partner'] = {
+                'id': -1,
+                'email': email,
+                'name': name,
+                'enrichment_info': None,
+            }
+            company = self._find_existing_company(normalized_email)
+            if not company:  # create and enrich company
+                company, enrichment_info = self._create_company_from_iap(sender_domain)
+                response['partner']['enrichment_info'] = enrichment_info
+            response['partner']['company'] = self._prepare_company_values(company)
+
+        return response
+
+    @http.route('/mail_plugin/partner/search', type="json", auth="outlook", cors="*")
+    def res_partners_search(self, search_term, limit=30, **kwargs):
+        """
+        Used for the plugin search contact functionality where the user types a string query in order to search for
+        matching contacts, the string query can either be the name of the contact, it's reference or it's email.
+        We choose these fields because these are probably the most interesting fields that the user can perform a
+        search on.
+        The method returns an array containing the dicts of the matched contacts.
+        """
+
+        #In a multi-company environment, the method may return contacts not belonging to the company that the user
+        #is connected to, this may result in the user not being able to view the contact in Odoo, while this may happen
+        #it is not supported for now and users are encouraged to check if they are connected to the correct company before
+        # clicking on a contact.
+        filter_domain = ['|', '|', ('display_name', 'ilike', search_term), ('ref', '=', search_term),
```

Comment:
Why searching on display_name and not name ?

Reply:
Since what's shown in the Odoo interface is the display_name and not the name, we can expect the user to run queries using the display_name, for this reason, if we rely on the name itself the user can get confused.